FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/AzureAccessTokenProvider.cs

```diff
@@ -4,46 +4,57 @@
 // -------------------------------------------------------------------------------------------------
 
 using System;
+using System.Collections.Concurrent;
 using System.Threading;
 using System.Threading.Tasks;
+using Azure.Core;
 using EnsureThat;
-using Microsoft.Azure.Services.AppAuthentication;
 using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Synapse.Common.Authentication;
 
 namespace Microsoft.Health.Fhir.Synapse.DataClient.Api
 {
+    /// <summary>
+    /// Provide access token for the given FHIR url from token credential.
+    /// Access token will be cached and will be refreshed if expired.
+    /// </summary>
     public class AzureAccessTokenProvider : IAccessTokenProvider
     {
-        private readonly AzureServiceTokenProvider _azureTokenProvider;
         private readonly ILogger<AzureAccessTokenProvider> _logger;
+        private readonly TokenCredential _tokenCredential;
+        private ConcurrentDictionary<string, AccessToken> _accessTokenDic = new ();
+        private const int _tokenExpireInterval = 5;
 
-        public AzureAccessTokenProvider(ILogger<AzureAccessTokenProvider> logger)
+        public AzureAccessTokenProvider(ITokenCredentialProvider credentialProvider, ILogger<AzureAccessTokenProvider> logger)
         {
+            EnsureArg.IsNotNull(credentialProvider, nameof(credentialProvider));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
-            _azureTokenProvider = new AzureServiceTokenProvider();
             _logger = logger;
+            _tokenCredential = credentialProvider.GetCredential(TokenCredentialTypes.External);
```

Comment:
Do we need hard code the credential type here?

Reply:
Updated it.